### PR TITLE
Easy tck

### PR DESCRIPTION
**Pull request by folker-kuhn**
Make it easier to run single TCK tests. 

Usage

mvn install -Drun.test="com/sun/ts/tests/servlet/spec/annotationservlet/webservletdd/URLClient.java#test1"
mvn install -Drun.test="com/sun/ts/tests/servlet/spec/annotationservlet/webservletdd/URLClient.java#test1" -Dsuspend

Future extension could include patterns easily. The script would work even better when Payara would write a file when started on which the script can do the wait instead of the time out.

### CODE REVIEW

**Comment by arjantijms** [comment]
We need to add an indicator that Piranha has started indeed. I'll take a look at that whenever I have some time.  The ts-failed.jtx would maybe be more useful if it did the reverse; only running those tests that failed.

I vaguely remember the TCK already had a switch for that somewhere though.

**Review by arjantijms — APPROVED** [review]
LGTM, we do need to keep the failed file updated when we improve the container, and check if new excludes and/or tests are added when we update the TCK version.

File: external/tck/servlet/pom.xml [hunk 1]
```diff
@@ -36,10 +36,10 @@
     
     <profiles>
         <profile>
-            <id>linux</id>
+            <id>unix</id>
```
Comment (mnriem) [inline external/tck/servlet/pom.xml:39]: Please do not change this as this is a 1:1 mapping for GitHub workflow

File: external/tck/servlet/pom.xml [hunk 2]
```diff
@@ -408,7 +461,7 @@
                                     <artifactItems>
                                         <artifactItem>
                                             <groupId>cloud.piranha</groupId>
-                                            <artifactId>piranha-server</artifactId>
+                                            <artifactId>piranha-server2</artifactId>
```
Comment (mnriem) [inline external/tck/servlet/pom.xml:464]: Why did this change?

File: external/tck/servlet/src/test/etc/ts-all.jtx [hunk 3]
```diff
@@ -0,0 +1,1735 @@
+#
```
Comment (mnriem) [inline external/tck/servlet/src/test/etc/ts-all.jtx:1]: Why are you creating this?

File: external/tck/servlet/src/test/etc/ts-failed [hunk 4]
```diff
@@ -0,0 +1,142 @@
+#
```
Comment (mnriem) [inline external/tck/servlet/src/test/etc/ts-failed:1]: Why are you creating this?

File: external/tck/servlet/src/test/etc/ts.jtx [hunk 5]
```diff
@@ -0,0 +1,52 @@
+#
```
Comment (mnriem) [inline external/tck/servlet/src/test/etc/ts.jtx:1]: Why are you creating this?

**Comment by mnriem** [comment]
As the current cycle is winding down this change is too large to be just be merged. And it warrants more discussion as it change so many things in one PR

**Comment by mnriem** [comment]
Also please create an issue linked to this PR

**Comment by mnriem** [comment]
@folker-kuhn You did nothing wrong. The timing of the TCK change is a bit inconvenient as tonight / tomorrow all the TCKs are run and as the release manager for the release I am a bit uncomfortable with such a large change at the 11th hour. I apologize for not clearly communicating this until now. Note that the release is on the 13th of every month so I most definitely want to revisit this with some additional discussion after the release.

**Comment by mnriem** [comment]
See #1969

**Pull request merged**